--- linux-3.3.7.orig/drivers/ssb/driver_chipcommon.c	2012-05-21 21:42:51.000000000 +0300
+++ linux-3.3.7/drivers/ssb/driver_chipcommon.c	2012-06-02 00:43:30.530807365 +0300
@@ -475,7 +475,8 @@ int ssb_chipco_serial_init(struct ssb_ch
 		} else if ((ccrev >= 11) && (ccrev != 15)) {
 			/* Fixed ALP clock */
 			baud_base = 20000000;
-			if (cc->capabilities & SSB_CHIPCO_CAP_PMU) {
+			if ((cc->capabilities & SSB_CHIPCO_CAP_PMU) &&
+			    bus->chip_id != 0x4716) {
 				/* FIXME: baud_base is different for devices with a PMU */
 				SSB_WARN_ON(1);
 			}
